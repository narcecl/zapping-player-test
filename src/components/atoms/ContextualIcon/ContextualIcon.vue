<template>
    <div class="contextual_icon">
        <slot />
    </div>
</template>

<style lang="scss">
.contextual_icon {
    align-items: center;
    background-color: rgba(white, 0.1);
    border-radius: 4px;
    display: inline-flex;
    padding: 0 12px;
    height: 24px;
    justify-content: center;
    transition: all 0.3s ease;
    border: 1px solid rgba(white, 0.1);
    color: white;
    font-size: 0.625rem;
}
</style>
