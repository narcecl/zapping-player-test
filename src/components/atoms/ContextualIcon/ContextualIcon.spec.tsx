import { describe, test, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import ContextualIcon from './ContextualIcon.vue';

describe('Badge', () => {
    test('Default render (button)', () => {
        const wrapper = mount(<ContextualIcon>F</ContextualIcon>);
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain('F');
        expect([...wrapper.element.classList]).includes('contextual_icon');
    });
});
