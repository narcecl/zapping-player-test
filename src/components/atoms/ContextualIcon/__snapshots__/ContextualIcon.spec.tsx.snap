// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > Default render (button) 1`] = `"<div class="contextual_icon">F</div>"`;
